pkg/installer/custominstallconfig.go

```diff
@@ -147,12 +162,117 @@ func (m *manager) applyInstallConfigCustomisations(installConfig *installconfig.
 	if err != nil {
 		return nil, err
 	}
+	dnsmasqMasterMachineConfig, err := dnsmasq.MachineConfig(installConfig.Config.ClusterDomain(), dnsConfig.APIIntIP, dnsConfig.IngressIP, "master", dnsConfig.GatewayDomains, dnsConfig.GatewayPrivateEndpointIP, true)
+	if err != nil {
+		return nil, err
+	}
+	data, err := yaml.Marshal(dnsmasqMasterMachineConfig)
+	if err != nil {
+		return nil, err
+	}
+	masterConfig := ignition.FileFromBytes("/opt/openshift/openshift/99_openshift-machineconfig_99-master-aro-dns.yaml", "root", 0644, data)
+	dnsmasqWorkerMachineConfig, err := dnsmasq.MachineConfig(installConfig.Config.ClusterDomain(), dnsConfig.APIIntIP, dnsConfig.IngressIP, "worker", dnsConfig.GatewayDomains, dnsConfig.GatewayPrivateEndpointIP, true)
+	if err != nil {
+		return nil, err
+	}
+	data, err = yaml.Marshal(dnsmasqWorkerMachineConfig)
+	if err != nil {
+		return nil, err
+	}
+	workerConfig := ignition.FileFromBytes("/opt/openshift/openshift/99_openshift-machineconfig_99-worker-aro-dns.yaml", "root", 0644, data)
```

Comment:
Does the part above here belong to the previous patch?
This code is repetitive and looks like it could be refactored into a function. Preferably in the dnsmasq package.